### ISSUE TITLE
Useful sample request...

### ISSUE THREAD

**Issue by Perksey** [mellinoe/veldrid-samples#3]
Not saying that the other samples were bad or anything, but shouldn't there be a sample explaining how one would go about making the scene that seems to be advertised everywhere in this project?

![image](https://user-images.githubusercontent.com/11160611/39677967-f0270630-517b-11e8-98f4-84b06e935df2.png)

**Comment by mellinoe** [comment]
There are certainly a lot of holes in the sample projects currently, especially in the beginner area. I'm hoping to be able to spend some more time on this in the near future, and to write more complete tutorials / blogs about using Veldrid.

By the way, those images are rendered from the [NeoDemo](https://github.com/mellinoe/veldrid/tree/master/src/NeoDemo) project over in the Veldrid repo. While it uses some nice, high-quality techniques, the code is fairly messy and disorganized. So it is not ideal as a sample project. I use it mainly as a more complete test program for things that can't be easily unit-tested. I would like for the techniques used there to be easily understood from the projects here in this repo, though.

**Comment by DylanFPS** [comment]
Yeah... If you like I can do an automated code cleanup and PR (on the main
repo)

**Comment by mellinoe** [comment]
It's not really a matter of simple code cleanup, I think. The code style/formatting is fine, and that's all you could reliably fix with an automated tool. NeoDemo just isn't structured in a way that is good for a "sample project". It's a messy grab bag of tons of features in lots of random places, because I wanted to get good coverage, and I've changed it a bunch over time as Veldrid itself evolved. It would probably be better to have several projects, or a tutorial series, that work up to the fulfeature set used in NeoDemo. Like:

1. How to render an obj file
2. How to set up a scene with many obj files, with frustum culling
3. How to draw alpha-blended objects (sponza foliage)
4. How to draw a directional light (no shadows)
5. How to draw a skybox
6. How to draw directional shadows
7. How to draw point lights
8. How to use cascaded shadow maps
9. How to use MSAA
10. How to draw planar reflections

**Comment by DylanFPS** [comment]
Yeah defo, cause right now I can't seem to process the NeoDemo program XD

A tutorial series on this would definitely clear things up.
